File: src/main/java/ar/com/flow/bankaccount/withdrawal/SufficientFunds.java
package ar.com.flow.bankaccount.withdrawal;

import ar.com.flow.bankaccount.BankAccount;
import ar.com.flow.bankaccount.transaction.Precondition;
import ar.com.flow.money.InsufficientFundsException;
import ar.com.flow.money.Money;
import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
public class SufficientFunds implements Precondition {
    private final BankAccount account;
    private final Money amount;

    public void check() {
        if (!account.withdrawalLimitSupports(amount)) {
            throw new InsufficientFundsException();
        }
    }
}

File: src/main/kotlin/ar/com/flow/bankaccount/withdrawal/SufficientFunds.kt
package ar.com.flow.bankaccount.withdrawal

import ar.com.flow.bankaccount.BankAccount
import ar.com.flow.bankaccount.transaction.Precondition
import ar.com.flow.money.InsufficientFundsException
import ar.com.flow.money.Money

class SufficientFunds(private val account: BankAccount, private val amount: Money) : Precondition {
    override fun check() {
        if (!account.withdrawalLimitSupports(amount)) {
            throw InsufficientFundsException()
        }
    }
}